imtest/imtest.c

```diff
@@ -1115,7 +1115,7 @@ static int auth_sasl(struct sasl_cmd_t *sasl_cmd, const char *mechlist)
         } else if (sendliteral) {
             /* If we had no response, we still need to send the
                empty literal in this case */
-            printf("{0+}\r\nC: ");
+            printf("C: {0+}");
```

Comment:
I don't know this code very well, but from looking at the other sections it looks like the "C: " needs to be conditional on initial_response.  I'm not sure if initial_response is already known to be false in this case.  Does something like this make sense?

```suggestion
            printf("%s{0+}", initial_response ? "" : "C: ");
```

Otherwise, `printf` with no format arguments is a bad smell, though I realise this line was already like that.  It's better to use `fputs`, like this:

```suggestion
            fputs("C: {0+}", stdout);
```

Reply:
> I don't know this code very well, but from looking at the other sections it looks like the "C: " needs to be conditional on initial_response. I'm not sure if initial_response is already known to be false in this case. Does something like this make sense?

Can you provide a real-world example with `imtest` or `sivtest`, where on that line it would be reasonable to hide `C:` by utilizing `initial_response`?

Reply:
I don't really use this tool, so I don't know of any invocation where the unconditional `C:` would cause problems, I'm only looking at what's different in the nearby code.  My understanding (perhaps incorrect) is that `initial_response` tracks whether for some reason we have already sent `C:` for this line, in which case we try to not send it again.  I don't know why we would have already sent `C:` for this line though.

You know this tool better than I do, and the change you've made is clearly better than what was already there.  So if you're not worried about `initial_response`, neither am I.  We can always fix it later if we do find a case where it's a problem.